clean.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "Usage: clean [options]"
+    echo "Cleans the local dev environment."
+    echo
+    echo "  -b     Delete bin directory"
+    echo "  -p     Delete packages directory and NuGet package caches"
+    echo "  -a     Remove all untracked files in the working tree and delete NuGet package caches"
```

Comment:
I don't have a strong preference for the working directory option. I suggested "-s" for source tree so it could apply to both TFS and git. The working tree is very much a git concept, but again I don't have a strong preference.